lib/base-controller.js

```diff
@@ -23,6 +23,57 @@ function getErrorLength() {
   }
 }
 
+/*
+ * Utility function which accepts an array of invalidating fields
+ * and the steps config object, returns a filtered steps object
+ * containing fields which have been invalidated.
+ */
+function getStepsToCheck(fields, steps) {
+  return _.pick(steps, function pickSteps(step) {
+    if (!step.fields) {
+      return false;
+    }
+    return _.intersection(step.fields, fields).length;
+  });
+}
+
+/*
+ * Utility function which accepts an array of invalidating fields,
+ * the steps config object and the sessionModel. It returns an array
+ * of step names for which all fields have been invalidated.
+ */
+function getInvalidatedSteps(fields, steps, sessionModel) {
+  var stepsToCheck = getStepsToCheck(fields, steps);
+
+  return _.chain(stepsToCheck).pick(function pickSteps(step) {
+    return _.every(step.fields, function everyField(item) {
+      return sessionModel.get(item) === undefined;
+    });
+  }).keys().value();
+}
+
+BaseController.prototype.saveValues = function saveValues(req, res, next) {
+  this.invalidateSteps(req, res, next);
+  Controller.prototype.saveValues.apply(this, arguments);
+};
+
+/*
+ * Remove any steps from the sessionModel
+ * that have had all of their fields invalidated
+ */
+BaseController.prototype.invalidateSteps = function invalidateSteps(req) {
+  var steps = this.options.steps;
+  var invalidatingFields = _.pick(this.options.fields, function pickFields(f) {
+    return f && f.invalidates && f.invalidates.length;
+  });
+  _.each(invalidatingFields, function eachInvalidatingField(field, key) {
```

Comment:
@JoeChapman invalidatingFields is an Object not an Array, `_.each` works on collections so objects and arrays